### ISSUE TITLE
"Old school" error mode is now missing information first/second borrow

### ISSUE THREAD

**Issue by jonathandturner** [rust-lang/rust#33543]
From https://github.com/rust-lang/rust/issues/33240#issuecomment-218246575:

Using nightly (rustc 1.10.0-nightly (e0fd34bba 2016-05-09)) on this snippet https://is.gd/jP3UYZ has less helpful error messages than stable when not using RUST_NEW_ERROR_FORMAT? E.g.

```
x.rs:4:15: 4:16 error: cannot borrow `a` as mutable more than once at a time [E0499]
x.rs:4  let c = &mut a;
                     ^
x.rs:4  let b = &mut a;
                     ^
x.rs:4 }
       ^
```

Third, tabs seem to be broken on the old error messages? I'm pretty sure they used to work E.g.

```
src/main.rs:79:35: 79:53 error: failed to resolve. Use of undeclared type or module `mem` [E0433]
src/main.rs:79          Moveable { v: Box::new(unsafe { mem::uninitialized() }) }
                                                 ^~~~~~~~~~~~~~~~~~
```

**Comment by sanxiyn** [comment]
span_label is not handled in old format.

**Comment by nagisa** [comment]
NOTE: I added the code block formatting to the original report.

**Comment by jonathandturner** [comment]
This should be fixed with #33688
